angularjs-portal-home/src/main/webapp/partials/widgets/weather.html

```diff
@@ -25,6 +29,44 @@
           </a>
         </li>
     </ul>
+    
+    <div ng-if="weather.location == null">
+		<div ng-model="noLocations" ng-init="noLocations=true">
+			<i class="fa fa-exclamation-triangle" aria-label='Warning'></i>
+			<div class="warning-message">Hold up! You haven't set any weather locations yet! Launch the full app to select your weather locations.</div>
+			<style>
+			
+				.fa-exclamation-triangle{
+					font-size: -webkit-xxx-large;
+					padding: 20px 120px;
+				}
+				.warning-message{
+					text-align:-webkit-center;
+					padding: 9px 0px;
+				}
+				
+			</style>
+		</div>
+	</div>
+    
+    <div ng-if="weatherData.length == 0 && noLocations==false">
+	    <i class="fa fa-frown-o" aria-label='Error'></i>
```

Comment:
Nitpick : consistent use of `"` or `'` to quote attributes within an element is preferable.